### PR TITLE
Ngram models

### PR DESCRIPTION
**Pull request by yaya-sy**

### CODE REVIEW

File: scripts/models/ngram_model/ngram_lm.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+import pickle
+import logging
+from pathlib import Path
+from itertools import tee
+from collections import defaultdict
+
+# import downloaded packages
+import numpy as np
+
+random.seed(1798)
```
Comment (LoannPeurey) [inline scripts/models/ngram_model/ngram_lm.py:16]: What is the purpose of setting the seed, and to that particular value? I could not find any reference to random in the files after this

File: scripts/models/ngram_model/ngram_lm.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    ---------
+    - smooth: float, int
+        Value for smoothing the probability distributions\
+        for dealing with unknown units.
+    - pad_utterances: bool
+        Whether add or not fake tokens at the begenning and ending of the utterance.
+    - parameters: defaultdict
+        Will contain the ngrams counts.
+    """
+    def __init__(self, smooth: Union[float, int]=1e-3):
```
Comment (LoannPeurey) [inline scripts/models/ngram_model/ngram_lm.py:37]: Correct me if I am wrong, the indication that smooth should be int or float is purely for documentation and indicating that both types can be used?

File: scripts/models/ngram_model/ngram_lm.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
+"""This module implements a ngram language model."""
```
Comment (LoannPeurey) [inline scripts/models/ngram_model/ngram_lm.py:1]: This is just a detail but we could provide more info on what is included here (perhaps that you can use Ngram or UniGram, what you can do with them etc)

**Review by LoannPeurey — COMMENTED** [review]
(applies only to scripts/models/ngram_model/ngram_lm.py)
Good global structure and usage of class, inheritance and abstract methods
Good level of documentation, docstrings etc

File: scripts/models/ngram_model/ngram_lm.py (hunk 1 above)
Comment (yaya-sy) [reply scripts/models/ngram_model/ngram_lm.py:16]: Yes in fact, you're right, this line of code is not necessary

File: scripts/models/ngram_model/ngram_lm.py (hunk 2 above)
Comment (yaya-sy) [reply scripts/models/ngram_model/ngram_lm.py:37]: yes, the smooth value is in range ]0;1], so can also be 1 (an int)

**Pull request closed without merging**